zipline/gens/brokers/alpaca_broker.py

```diff
@@ -0,0 +1,261 @@
+        side = 'buy' if amount > 0 else 'sell'
+        order_type = 'market'
+        if isinstance(style, MarketOrder):
+            order_type = 'market'
+        elif isinstance(style, LimitOrder):
+            order_type = 'limit'
+        elif isinstance(style, StopOrder):
+            order_type = 'stop'
+        elif isinstance(style, StopLimitOrder):
+            order_type = 'stop_limit'
+
+        limit_price = style.get_limit_price(side == 'buy') or None
+        stop_price = style.get_stop_price(side == 'buy') or None
+
+        zp_order_id = self._new_order_id()
+        dt = pd.to_datetime('now', utc=True)
+        zp_order = ZPOrder(
+            dt=dt,
+            asset=asset,
+            amount=amount,
+            stop=stop_price,
+            limit=limit_price,
+            id=zp_order_id,
+        )
+
+        order = self._api.submit_order(
+            symbol=symbol,
+            qty=qty,
+            side=side,
+            type=order_type,
+            time_in_force='day',
+            limit_price=limit_price,
+            stop_price=stop_price,
+            client_order_id=zp_order.id,
+        )
+        zp_order = self._order2zp(order)
+        return zp_order
+
+    @property
+    def orders(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders()
+        }
+
+    @property
+    def transactions(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders(status='all')
+        }
+
+    def cancel_order(self, zp_order_id):
+        try:
+            order = self._api.get_order_by_client_order_id(zp_order_id)
+            self._api.cancel_order(order.id)
+        except Exception as e:
+            log.error(e)
+            return
+
+    def get_last_traded_dt(self, asset):
+        quote = self._api.get_quote(asset.symbol)
+        return pd.Timestamp(quote.last_timestamp)
+
+    def get_spot_value(self, assets, field, dt, data_frequency):
+        assert(field in (
+            'open', 'high', 'low', 'close', 'volume', 'price', 'last_traded'))
+        assets_is_scalar = not isinstance(assets, (list, set, tuple))
+        is_daily = 'd' in data_frequency
+        if assets_is_scalar:
+            symbols = [assets.symbol]
+        else:
+            symbols = [asset.symbol for asset in assets]
+        if field == 'price' or field == 'last_traded':
+            quotes = self._api.list_quotes(symbols)
+            if assets_is_scalar:
+                if field == 'price':
+                    return quotes[-1].last
+                else:
+                    return quotes[-1].last_timestamp
+            else:
+                return [
+                    quote.last if field == 'price' else quote.last_timestamp
+                    for quote in quotes
+                ]
+
+        timeframe = '1D' if is_daily else '1Min'
+        bars_list = self._api.list_bars(symbols, timeframe, limit=1)
```

Comment:
Out of curiosity:
Does this statement returns different values for 1D or 1Min time frame?
Since we're taking the last bar I'd expect it to return the same.

Reply:
It does return different result, yes. So, I should always use 1 min?  I wasn't sure what does expectedly mean "high" or "low" for "spot"

Reply:
always the last traded price

Reply:
OK then changing to 1Min only